File: examples/python/evm/send_transaction.py
# Usage: uv run python evm/send_transaction.py

import asyncio

from web3 import Web3

from cdp import CdpClient
from cdp.evm_transaction_types import TransactionRequestEIP1559
from dotenv import load_dotenv

load_dotenv()

w3 = Web3(Web3.HTTPProvider("https://sepolia.base.org"))


async def main():
    async with CdpClient() as cdp:
        account = await cdp.evm.create_account()
        print(f"Created account: {account.address}")

        faucet_hash = await cdp.evm.request_faucet(
            address=account.address, network="base-sepolia", token="eth"
        )

        w3.eth.wait_for_transaction_receipt(faucet_hash)
        print(f"Received funds from faucet for address: {account.address}")

        tx_hash = await cdp.evm.send_transaction(
            address=account.address,
            transaction=TransactionRequestEIP1559(
                to="0x0000000000000000000000000000000000000000",
                value=w3.to_wei(0.000001, "ether"),
            ),
            network="base-sepolia",
        )

        print(f"Transaction sent! Hash: {tx_hash}")

        print("Waiting for transaction confirmation...")
        tx_receipt = w3.eth.wait_for_transaction_receipt(tx_hash)
        print(f"Transaction confirmed in block {tx_receipt.blockNumber}")
        print(
            f"Transaction status: {'Success' if tx_receipt.status == 1 else 'Failed'}"
        )


asyncio.run(main())

File: examples/typescript/evm/sendTransaction.ts
// Usage: pnpm tsx evm/sendTransaction.ts

import { CdpClient } from "@coinbase/cdp-sdk";

import { parseEther, createPublicClient, http } from "viem";
import { baseSepolia } from "viem/chains";
import "dotenv/config";

const cdp = new CdpClient();
const publicClient = createPublicClient({
  chain: baseSepolia,
  transport: http(),
});

// Step 1: Create a new EVM account
const account = await cdp.evm.createAccount();
console.log("Successfully created EVM account:", account.address);

// Step 2: Request ETH from the faucet
const { transactionHash: faucetTransactionHash } = await cdp.evm.requestFaucet({
  address: account.address,
  network: "base-sepolia",
  token: "eth",
});

// Wait for the faucet transaction to be confirmed onchain.
const faucetTxReceipt = await publicClient.waitForTransactionReceipt({
  hash: faucetTransactionHash,
});

console.log(
  "Successfully requested ETH from faucet:",
  faucetTxReceipt.transactionHash
);

// Step 3: Sign and send the transaction in a single step with sendTransaction.
const txResult = await cdp.evm.sendTransaction({
  address: account.address,
  network: "base-sepolia",
  transaction: {
    to: "0x4252e0c9A3da5A2700e7d91cb50aEf522D0C6Fe8", // recipient address
    value: parseEther("0.000001"), // sending 0.000001 ETH
  },
});

console.log(
  `Transaction hash: ${txResult.transactionHash}. Waiting for confirmation...`
);

await publicClient.waitForTransactionReceipt({
  hash: txResult.transactionHash,
});

console.log(
  `Transaction confirmed! link: https://sepolia.basescan.org/tx/${txResult.transactionHash}`
);
